pkg/datamover/repository_server_test.go

```diff
@@ -0,0 +1,199 @@
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/param"
+)
+
+type RepositoryServerSuite struct {
+	namespace        *corev1.Namespace
+	pod              *corev1.Pod
+	service          *corev1.Service
+	ctx              context.Context
+	cli              kubernetes.Interface
+	s3Creds          *corev1.Secret
+	s3Location       *corev1.Secret
+	tlsSecret        *corev1.Secret
+	userAccessSecret *corev1.Secret
+	repoServer       *param.RepositoryServer
+}
+
+var _ = Suite(&RepositoryServerSuite{})
+
+func (rss *RepositoryServerSuite) SetUpSuite(c *C) {
+	// Set Context as Background
+	rss.ctx = context.Background()
+
+	// Get Kubernetes Client
+	cli, err := newTestClient()
+	c.Assert(err, IsNil)
+	rss.cli = cli
+
+	// Create Test Namespace
+	rss.namespace, err = createRepositoryServerTestNamespace(rss.ctx, rss.cli)
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test S3 Credential and Location Secrets
+	rss.s3Creds, rss.s3Location, err = testS3CredsLocationSecret(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test User Access Secret for Kopia Repository Server
+	rss.userAccessSecret, err = testKopiaRepositoryServerUserAccess(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Test TLS Certificate Secret
+	rss.tlsSecret, err = testKopiaTLSCertificate(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Create Test Pod
+	rss.pod, err = createRepositoryServerTestPod(rss.ctx, rss.cli, rss.namespace.GetName(), rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Wait for Test Pod to get Ready
+	c.Assert(kube.WaitForPodReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod.Name), IsNil)
+
+	// Create Test Service
+	rss.service, err = createRepositoryServerTestService(rss.ctx, rss.cli, rss.namespace.GetName())
+	c.Assert(err, IsNil)
+
+	// Configure and Create Kopia Repository
+	err = createTestKopiaRepository(rss.s3Location, rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Start Kopia Repository Server
+	err = startTestKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Add Test Client User in Kopia Repository
+	err = addTestUserInKopiaRepositoryServer(rss.cli, rss.namespace.GetName(), rss.pod)
+	c.Assert(err, IsNil)
+
+	// Refresh Kopia Repository Server
+	err = refreshTestKopiaRepositoryServer(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
+
+	// Wait for Kopia Repository Server To Get Ready
+	err = waitForServerReady(rss.ctx, rss.cli, rss.namespace.GetName(), rss.pod, rss.tlsSecret)
+	c.Assert(err, IsNil)
```

Comment:
Also, is it possible to use kopia server setup methods we have already implemented in controller workflow?

Reply:
Since we're not creating Repository Server CR in this workflow and Controller Workflow heavily depends on it. We can not use the existing workflow here